java/ingest/ingest-batcher-store/src/main/java/sleeper/ingest/batcher/store/DynamoDBIngestBatcherStore.java

```diff
@@ -93,21 +94,24 @@ public void addFile(FileIngestRequest fileIngestRequest) {
 
     @Override
     public void assignJob(String jobId, List<FileIngestRequest> filesInJob) {
-        dynamoDB.transactWriteItems(new TransactWriteItemsRequest()
-                .withTransactItems(filesInJob.stream()
-                        .flatMap(file -> Stream.of(
-                                new TransactWriteItem().withDelete(new Delete()
-                                        .withTableName(requestsTableName)
-                                        .withKey(DynamoDBIngestRequestFormat.createUnassignedKey(file))
-                                        .withConditionExpression("attribute_exists(#filepath)")
-                                        .withExpressionAttributeNames(Map.of("#filepath", FILE_PATH))),
-                                new TransactWriteItem().withPut(new Put()
-                                        .withTableName(requestsTableName)
-                                        .withItem(DynamoDBIngestRequestFormat.createRecord(
-                                                tablePropertiesProvider, file.toBuilder().jobId(jobId).build()))
-                                        .withConditionExpression("attribute_not_exists(#filepath)")
-                                        .withExpressionAttributeNames(Map.of("#filepath", FILE_PATH))))
-                        ).collect(Collectors.toList())));
+        for (int i = 0; i < filesInJob.size(); i += BATCH_SIZE) {
+            List<FileIngestRequest> filesInBatch = filesInJob.subList(i, Math.min(i + BATCH_SIZE, filesInJob.size()));
+            dynamoDB.transactWriteItems(new TransactWriteItemsRequest()
```

Comment:
This isn't quite good enough, as the update needs to be atomic. If we split it into batches, we can have a state where the first batch passes but the second batch fails. The job won't be created but the first batch is still updated in DynamoDB.

We probably need to change the logic of the batcher to stop it creating jobs too big to fit in a transaction.

Reply:
Thanks, I've adjusted the logic to throw a custom exception if the file count is too large

Reply:
That's the same behaviour as we had before, except that it won't go to DynamoDB. Just making a note that we updated the issue together with a couple of options for how to fix it (https://github.com/gchq/sleeper/issues/1120). We planned to go with splitting up the job, by changing the validation for `sleeper.default.ingest.batcher.job.max.files` so that it can't go higher than 50.